### PR TITLE
fix: use pipeline to pipe lassie response through metrics

### PR DESCRIPTION
**Pull request by rvagg**
pipeline() will wire up error events properly; specifically in this case an abort from node-fetch won't propagate through metricsStream and will be lost entirely, not even bubbling up as an error through respondFromLassie().

There's an error callback that could be used here, perhaps for logging? But for now at least the error will come back through the CarBlockIterator for-await and back up through the call stack.

### CODE REVIEW

**Comment by guanzo** [comment]
Thanks, completely forgot about error handling on the `pipe`, will review in the mornin'.

File: container/shim/src/fetchers/lassie.js [hunk 1]
```diff
@@ -105,13 +105,13 @@ export async function respondFromLassie(req, res, { cidObj, format }) {
     } else {
       res.set("Cache-Control", "public, max-age=29030400, immutable");
     }
-    const carStream = lassieRes.body.pipe(metricsStream);
+    pipeline(lassieRes.body, metricsStream, (err) => {});
```
Comment (juliangruber) [inline container/shim/src/fetchers/lassie.js:108]: Just double checking, do we want to not log errors here?

**Review by juliangruber — APPROVED** [review]

File: container/shim/src/fetchers/lassie.js (hunk 1 above)
Comment (guanzo) [reply container/shim/src/fetchers/lassie.js:108]: We do. That's something I can implement post merge

Comment (guanzo) [reply container/shim/src/fetchers/lassie.js:108]: After testing, errors are caught by the below `catch` block. An empty catch handler seems fine here?

Comment (rvagg) [reply container/shim/src/fetchers/lassie.js:108]: yeah, that's the point of the empty callback - those errors end up coming back through the async iterator from js-car anyway; but now you have multiple options! not confusing at all

Comment (guanzo) [reply container/shim/src/fetchers/lassie.js:108]: Yep i added one. Also worth noting that if `pipeline` throws an error if the last argument isn't a callback, so you have to pass one.

Comment (rvagg) [reply container/shim/src/fetchers/lassie.js:108]: fwiw you could also just grab the version from `'node:stream/promises'` and drop the returned Promise, no callback needed. This is more explicit I think though.

**Pull request merged**